### PR TITLE
fix #7100: remove `@theia/languages` and `monaco-languageclient`

### LINKED ISSUE #7100: remove dependency to monaco-languageclient
Right now it is quire hard to catch up with Monaco because we have to upgrade monaco-languageclient each time and make sure that translation between vscode api and lsp works in the browser context as well as vscode-languageclient (which actually written for Node.js).

Since we have possibility to run language servers as part of vscode extensions. I suggest that we simplify and get rid of monaco-languageclient. A consequence is that existing Theia extensions contributing language features then should be rewritten as VS Code extensions.

After that process of upgrading to latest Monaco will look like:
- publish not-tree shaken version of monaco from theia-ide/vscode
- bump up a version in Theia and test it

I hope we will be able to add integration tests for all exposed internal Monaco apis and ideally move to nightly builds and testing of not-tree shaken monaco against Theia soon.

cc @svenefftinge @marcdumais-work @eclipse-theia/ecd-theia-committers 

@RomanNikitenko @azatsarynnyy it is not something to work on right now.

### PR DESCRIPTION
**Pull request by akosyakov**
#### What it does

- fix #7100: remove `@theia/languages` and `monaco-languageclient` - It allows to consume Monaco directly without trying to run vscode-languageclient in the browser. One should contribute language smartness via VS Code extensions instead.
- It cannot be part of 1.4.0, at least requires 1.5.0

TODO: 
- [x] following services should be deprecated in 1.4.0, see https://github.com/eclipse-theia/theia/pull/8139:
  - Language -> use `Language` from `@theia/core/lib/browser/language-service`
  - Languages -> use `monaco.languages.register*`, `MonacoLanguages.registerWorkspaceSymbolProvider` or `LanguageService` from `@theia/core/lib/browser/language-service`
  - Workspace -> use `MonacoWorkspace`
  - semantic highlighting -> VS Code extension after https://github.com/eclipse-theia/theia/pull/8010 and https://github.com/eclipse-theia/theia/issues/7608
  - call hierarchy -> VS Code extension
  - type hierarchy -> `TypeHierarchyRegistry.register` for now, later VS Code extension: https://github.com/microsoft/vscode/issues/15533
  - `languageServer` namespace -> use vscode-langaugeserver via VS Code extension
- [x] fix tests

#### How to test

- [x] workspace symbols provided by VS Code extensions should be available:
  - wait till ts is initialized after opening a file
  - trigger `Go to Workspace Symbol...` command
  - type some prefix, wait till symbols are found
  - select some symbol and navigate to it
- [x] suggestion in the debug console should be properly provided and applied
  - put a breakpoint in uri.spec.ts
  - debug with `Run Mocha Test`
  - use the debug console to lookup variables
  - use different prefixes, apply suggestions and check that they resolve to proper variable names
- [x] language status bar entry should work
  - open some file
  - try to change a language
  - try to auto detect
- [x] git blame hover should work
  - open some file
  - trigger blame annotations
  - hover over them and check that hover corresponds to commit in the ruler
- [x] markdown preview widget should be in sync with text editors as before
  - open several markdown files and preview them (make sure to use preview from Theia, not built-in extension, one with old icon)
  - check that scrolling between preview and editor in sync
  - check that dirty changes are reflected
  - check that closed dirty editor without save reverts dirty changes in the preview
- [x] call hierarchies contributed by VS Code extensions should work
  - open ts file and wait till ts server is ready
  - trigger `Open Call Hierarchy` on some method
- [x] problem markers should be properly reflected
  - open some ts file
  - break it, should see markers in the problems view
  - delete it, markers should be removed
  - revert it in the scm view, markers should be added back
  - revert the dirty changes, markers should be removed

#### Review checklist

- [x] as an author, I have thoroughly tested my changes and carefully followed [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#requesting-a-review)

#### Reminder for reviewers

- as a reviewer, I agree to behave in accordance with [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#reviewing)

### CODE REVIEW

**Comment by akosyakov** [comment]
@tsmaeder @benoitf How do we deal with `languageServer` namespace? We don't want to add a dependency to a concrete version of vscode-languageserver in the plugin system. Should it be the same use VS Code extension instead? Do you use it in practice?

**Comment by tsmaeder** [comment]
@akosyakov I'm not aware of anyone using the languageserver namespace. AFAIC go ahead and kill it.

**Comment by akosyakov** [comment]
It's ready, but I don't think we need to review it yet. More important now to help with landing https://github.com/eclipse-theia/theia/pull/8010 and fixing https://github.com/eclipse-theia/theia/issues/7608 in this month. Otherwise everything will be postponed for one month.

File: packages/monaco/src/browser/monaco-workspace.ts [hunk 1]
```diff
@@ -130,14 +116,7 @@ export interface WorkspaceFolder {
...
-    readonly capabilities = {
-        applyEdit: true,
-        workspaceEdit: {
-            documentChanges: true
-        }
-    };
+export class MonacoWorkspace {
 
     protected resolveReady: () => void;
     readonly ready = new Promise<void>(resolve => {
```
Comment (akosyakov) [inline packages/monaco/src/browser/monaco-workspace.ts:122]: it should be removed

**Comment by RomanNikitenko** [comment]
I rebased and fixed conflicts for the PR as was discussed https://github.com/eclipse-theia/theia/pull/8010#issuecomment-666341536

I'm going to wait for ci build/tests and provide light manual testing tomorrow morning before merge the PR.

**Comment by kittaakos** [comment]
> So, I'm going to merge the PR

Can we wait a few more days, please?

**Comment by benoitf** [comment]
hi @kittaakos, could you explain why you need more days ?

We just have a fresh tagged release so I think it's ok to merge all major changes in repo after this tag

like drop monaco-language-client, update FS api, bump monaco, etc.

**Comment by kittaakos** [comment]
> hi @kittaakos, could you explain why you need more days ?

All downstream projects still using language contributions from Theia cannot use the `next` version anymore.

**Comment by benoitf** [comment]
@kittaakos I'm not sure few more days would help then ? as basically at some point `next` will still remove the support of `monaco-language-client` as target is for 1.5.0
I mean, if it's dropped on monday, it will still break on monday.

**Comment by kittaakos** [comment]
> I mean, if it's dropped on monday, it will still break on monday.

I still have time to code until Monday.

**Comment by benoitf** [comment]
ok so let's wait until monday @RomanNikitenko ?

**Comment by RomanNikitenko** [comment]
I would like to land https://github.com/eclipse-theia/theia/pull/8010 ASAP
but, It's OK for me to wait until Monday if it helps to resolve the problem downstream projects

**Comment by kittaakos** [comment]
> to resolve the problem for downstream projects

Yes, it helps. For example, here is a Theia issue: https://github.com/eclipse-theia/theia/pull/8281
The only way to pick up such a fix by downstreams is by keeping the Theia version on `next`.

**Comment by benoitf** [comment]
@kittaakos we could do some 1.4 bugfixes release  as well (we did some of them in the past)

**Comment by kittaakos** [comment]
> I would like to land #8010 ASAP
> but, It's OK for me to wait until Monday

@RomanNikitenko, go for it. Thanks for your patience.

**Pull request merged**